msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMSALController.java

```diff
@@ -169,16 +177,51 @@ private Map getSilentParameters(MSALAcquireTokenSilentOperationParameters parame
         request.setForceRefresh(parameters.getForceRefresh());
         request.setLoginHint(parameters.getAccount().getUsername());
         request.setName(parameters.getAccount().getUsername());
-        request.setUserId(parameters.getAccount().getHomeAccountId());
+        request.setUserId(parameters.getAccount().getLocalAccountId());
         //request.setPrompt(parameters.get);
         //TODO: This should be the broker redirect URI and not the non-broker redirect URI
         request.setRedirect(parameters.getRedirectUri());
         //NOTE: Assumption: Broker will handle removing empty string scopes and appending stanard OIDC scopes
         request.setScope(StringUtil.join(' ', parameters.getScopes()));
+        request.setApplicationName(parameters.getAppContext().getPackageName());
```

Comment:
Setting this value 2x?

Reply:
Thanks for catching this. Changed as needed